src/main/java/seedu/address/logic/parser/ContactCommandParser.java

```diff
@@ -0,0 +1,39 @@
+package seedu.address.logic.parser;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import seedu.address.logic.commands.AddContactCommand;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+
+/**
+ * Parses input arguments and creates the appropriate ContactCommand objects.
+ */
+public class ContactCommandParser implements Parser<Command> {
+    public static final String COMMAND_WORD = "contact";
+    private static final Pattern CONTACT_COMMAND_FORMAT = Pattern.compile("(?<subCommand>\\S+)(?<subArguments>.*)");
+
+    @Override
+    public Command parse(String args) throws ParseException {
+        final Matcher matcher = CONTACT_COMMAND_FORMAT.matcher(args.trim());
+
+        if (!matcher.matches()) {
+            throw new ParseException("Invalid contact command format. Please refer to the help for usage.");
```

Comment:
Avoid string literals perhaps can use
```suggestions
            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
```